application/core/LSYii_ImageValidator.php

```diff
@@ -73,8 +73,14 @@ public static function validateImage($file)
             $result['uploadresult'] = '';
             $result['check'] = true;
         } else {
-            $result['uploadresult'] = sprintf(gT("This file is not a supported image format - only the following ones are allowed: %s"),strtoupper(Yii::app()->getConfig('allowedthemeimageformats')));
-            $result['check'] = false;
+            // If $checkImage is empty, it's most probably because fileinfo is missing. But we check it to be sure.
+            if (is_null($checkImage) && !function_exists('finfo_open')) { 
```

Comment:
> file_info are not required to get image type

It is not required, you are right. 
Still, if no "image type" found, one of the reasons could be not having the file_info.
That's why, after getting an "image type not found", we specifically check for file_info.

Not sure if I answered as expected or if we are talking about different stuff.

Thanks!

Reply:
> 

I think we could do a custom message for checking "mime_content_type" module as well.
Was that your thought?

Thanks!

Reply:
My opinion : 

1. Check if find mimetype of `APPPATH . "favicon.ico"` , don't test file_info or mime_content_type or who_care
2. In error message : tell one possibility ois fileinfo : something like «Unable to check format of the file, please activate file_info or mime_content_type» (but i'm not good in english.

Same for Installer requirements.

Then when a new function was added for my_file_getcontent : error message are bad, but test is OK ;)

Reply:
I use `APPPATH . "favicon.ico"` in uploader controller because

1. I'm sure we have it on all installation
2. we know their exact file_type

Then : if we don't get the file type : we don't have any way to get other file type.

Reply:
> 1. Check if find mimetype of `APPPATH . "favicon.ico"` , don't test file_info or mime_content_type or who_care

`CFileHelper::getMimeType` may check type by extension. That's why we explicitly check for the module.
https://github.com/LimeSurvey/LimeSurvey/blob/012a9ffd365dde56d20c0bc5ff176b2bd2353f07/framework/utils/CFileHelper.php#L275

Reply:
There are a parameters to use extension or no : LSFileHelper::getMimeType(APPPATH . "favicon.ico", null, **null**)

https://github.com/LimeSurvey/LimeSurvey/blob/012a9ffd365dde56d20c0bc5ff176b2bd2353f07/application/controllers/UploaderController.php#L209

Reply:
Yes : it's what i think :) .

Just replace `!function_exists('finfo_open')` by `empty(LSFileHelper::getMimeType(APPPATH . "favicon.ico", null, null))` and the sentence

Reply:
We have reviewed this again. What's the point in replacing a specific message that will tell a specific module has a problem by a message which is generic and give you less information?

Sorry, not sure I support this plan B.